src/ros2cs/rosidl_generator_cs/cmake/rosidl_generator_cs_generate_interfaces.cmake

```diff
@@ -223,15 +235,116 @@ foreach(_generated_msg_c_ts_file ${_generated_msg_c_ts_files})
     ${PROJECT_NAME}__rosidl_generator_c
   )
 
-  set(ros2_distro "$ENV{ROS_DISTRO}")
+  rosidl_target_interfaces(${_target_name}
```

Comment:
`rosidl_target_interfaces` is deprecated since Humble: https://docs.ros.org/en/humble/Releases/Release-Humble-Hawksbill.html#deprecation-of-rosidl-target-interfaces

It is a good idea to keep humble and up in check.